.github/workflows/release.yml

```diff
@@ -0,0 +1,203 @@
+          git config user.email 'kiali-dev@googlegroups.com'
+          
+          git config user.name 'kiali-bot'
+
+      - name: Build Helm charts
+        run: make -e VERSION=$RELEASE_VERSION clean build-helm-charts
+
+      # Switch to `master` branch before updating docs/index.yaml
+      - name: Checkout master
+        if: ${{ github.event.inputs.release_type == 'patch'}}
+        run: git checkout master
+      
+      # Anticipated preparation of Makefile so that it contains the released
+      # version when creating the git tag.
+      #
+      # This change to the Makefile is not done for patch releases, because that
+      # would break the next minor build (remember we have already switched to master).
+      - name: Prepare Makefile
+        if: ${{ github.event.inputs.release_type != 'patch'}}
+        run: sed -i -r 's/^VERSION \?= v.*/VERSION \?= v$RELEASE_VERSION/' Makefile
+
+      - name: Update Helm repositories
+        run: make -e VERSION=$RELEASE_VERSION update-helm-repos
+
+      # Tag the release from sources in the master branch
+      # Note that if we are doing a patch release, this tag won't contain valid `kiali-server` nor `kiali-operator` directories
+      # because these directories come from `master` rather than the original ones.
+      - name: Create tag
+        run: |
+          git add Makefile docs
+
+          git commit -m "Release $RELEASE_VERSION"
+          
+          git push origin $(git rev-parse HEAD):refs/tags/$RELEASE_VERSION-master
+    
+      - name: Publish Helm charts
+        env:
+          BUILD_TAG: helm-charts-release-${{ github.run_number }}-main
+        run: |
+          # If we did a minor release, we need to create the vX.Y branch, so that it can
+          # be used as a base for a patch release.
+          # Also, we create a vX.Y.Z tag.
+          if [[ $RELEASE_TYPE == "minor" ]]
+          then
+            git push origin $(git rev-parse HEAD):refs/heads/$BRANCH_VERSION
+            
+            git push origin $(git rev-parse HEAD):refs/tags/$RELEASE_VERSION
+          
+            # Bump version stored in the Makefile
+            sed -i -r "s/^VERSION \?= (.*)/VERSION \?= $NEXT_VERSION-SNAPSHOT/" Makefile
+
+            git add Makefile
+            
+            git commit -m "Prepare for next version"
+
+            git push origin $(git rev-parse HEAD):refs/heads/$BUILD_TAG
+
+            echo "{\"title\": \"Prepare for next version\", \"body\": \"Please, merge to update version numbers and prepare for release $NEXT_VERSION.\", \"head\": \"kiali-bot:$BUILD_TAG\", \"base\": \"master\"}"
+            curl -H "Authorization: token ${{ secrets.GH_TOKEN }}" \
+              -H "Content-Type: application/json" \
+              -d "{\"title\": \"Prepare for next version\", \"body\": \"Please, merge to update version numbers and prepare for release $NEXT_VERSION.\", \"head\": \"kiali-bot:$BUILD_TAG\", \"base\": \"master\"}" \
+              -X POST $GH_PULL_URI
```

Comment:
Like in the server side, perhaps use `gh` CLI for better clarity?

Reply:
Sure, I've already updated the server release pipeline to use gh cli. The rest of the pipelines will be modified soon.